frontend/src/pages/ImportCkan/SelectGroupsScreen.jsx

```diff
@@ -0,0 +1,589 @@
+          await (await fetch(`/api/MetadataType/${metadataTypeUuid}/tag`, {
+            method: 'PUT',
+            body: JSON.stringify({
+              name: tag.display_name.toLowerCase(),
+            }),
+            headers: {
+              'Content-Type': 'application/json',
+              Authorization: `bearer ${token}`,
+            },
+          })).json();
+        }
+      }
+      /* eslint-enable no-await-in-loop */
+
+      // Add the description if it exists
+      if (currentGroup.openDataMetadata.metadataTypeDescription.trim() !== '') {
+        await (await fetch(`/api/MetadataType/${metadataTypeUuid}/description`, {
+          method: 'PUT',
+          body: JSON.stringify({
+            content: currentGroup.openDataMetadata.metadataTypeDescription,
+          }),
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: `bearer ${token}`,
+          },
+        })).json();
+      }
+    } else {
+      // Append tags to the metadata type if they dont exist
+      // Create an array of existing tags
+      const typeTagList = metadataTypes[currentGroup.openDataMetadata.metadataType]
+        .tags
+        .map((tag) => tag.tagName.toLowerCase());
+
+      /* eslint-disable no-await-in-loop */
+      for (const tag of currentGroup.tags) {
+        // If the tag isn't already added on the metadata type
+        if (!typeTagList.includes(tag.display_name.toLowerCase())) {
+          await (await fetch(`/api/MetadataType/${currentGroup.openDataMetadata.metadataType}/tag`, {
+            method: 'PUT',
+            body: JSON.stringify({
+              name: tag.display_name.toLowerCase(),
+            }),
+            headers: {
+              'Content-Type': 'application/json',
+              Authorization: `bearer ${token}`,
+            },
+          })).json();
+        }
+      }
+      /* eslint-enable no-await-in-loop */
+    }
+
+    // Create metadata
+    let metadataUuid = currentGroup.openDataMetadata.metadata;
+    if (currentGroup.openDataMetadata.strategy <= 1) {
+      const createMetadataResult = await (await fetch('/api/Metadata', {
+        method: 'PUT',
+        body: JSON.stringify({
+          description: currentGroup.notes,
+          releaseState: 1, // Released (?)
+          metadataTypeUuid,
+          municipalityName,
+        }),
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `bearer ${token}`,
+        },
+      })).json();
+      metadataUuid = createMetadataResult.uuid;
+    }
+    // Insert metadata
+    /* eslint-disable no-await-in-loop */
+    for (const resource of currentGroup.resources) {
+      if (resource.openDataMetadata.ignore !== true) {
+        const payload = {
+          metadataUuid,
+          url: resource.url,
+          description: resource.description === '' ? '(No description)' : 'resource.description',
+          dataFormatMimeType: 'text/plain', // TODO start and end date
```

Comment:
Looks like start date and end date are implemented (on the lines below this)?
```suggestion
          dataFormatMimeType: 'text/plain',
```